### ISSUE TITLE
Add custom cursor(aggregation) on addFilterArg

### ISSUE THREAD

**Issue by 1k2k** [graphql-compose/graphql-compose#154]
Is there a way of adding custom mongoose cursors into an existing resolver, let's say the `connection` resolver, normally there is a way to add a custom `query` but at times we have aggregations not queries.

e.g 

```
OrderLineTC.getResolver('connection').addFilterArg({
  name: 'name',
  type: 'String',
  query: (rawQuery, value, resolveParams) => {
    rawQuery = // to another query.
  },
});
```

What if we had a query that is an aggregation, according to mongoose, db.collection.find & db.collection.aggregate both return `Cursors`, so why not override a cursor when finding documents.

e.g 
```
 query: (rawQuery, value, resolveParams) => {
    rawQuery = SomeModel.aggregate()....
  },
```

**Comment by 1k2k** [comment]
@nodkz Here's my current situation.

I already did a custom resolver and it worked, but i want keep the nice **pageInfo** provided by existing resolvers i.e `connection` or `pagination`.

Please correct me if am wrong, is it possible to override the `connection` or `pagination` resolvers so that i define my own cursors?

**Comment by nodkz** [comment]
Yep, `TC.setResolver()` method just replace existed resolver in your type. You just replace resolver under some name with a new instance with your logic.
